src/CitizenProject/CitizenProjectManager.php

```diff
@@ -498,14 +498,51 @@ public function hasCitizenProjectInStatus(Adherent $adherent, array $status): bo
         return $this->getCitizenProjectRepository()->hasCitizenProjectInStatus($adherent, $status);
     }
 
+    /**
+     * Uploads and saves the citizen project image to the cloud storage.
+     *
+     * @param CitizenProject $citizenProject
+     */
     public function addImage(CitizenProject $citizenProject): void
     {
-        // Save citizen project image to cloud storage
         if ($citizenProject->getImage() instanceof UploadedFile) {
-            $pathImage = $citizenProject->getImagePath();
-            $this->storage->put($pathImage, file_get_contents($citizenProject->getImage()->getPathname()));
-            $this->glide->deleteCache($pathImage);
+            $path = $citizenProject->getImagePath();
+
+            // Clears the old image if needed
+            if (null !== $citizenProject->getImageName() && $oldImagePath = $citizenProject->getImagePath()) {
+                $this->storage->delete($oldImagePath);
+            }
+
+            $citizenProject->setImageName($citizenProject->getImage());
+
+            // Uploads the file : creates or updates if exists
+            $this->storage->put($path, file_get_contents($citizenProject->getImage()->getPathname()));
+
+            // Clears the cache file
+            $this->glide->deleteCache($path);
+
             $citizenProject->setImageUploaded(true);
         }
     }
+
+    /**
+     * Removes the citizen project image from the cloud storage.
+     *
+     * @param CitizenProject $citizenProject
+     */
+    public function removeImage(CitizenProject $citizenProject): void
+    {
+        if ($citizenProject->isRemoveImage()) {
```

Comment:
It does not feel like it's the responsibility of the manager to test this. IMO, the entity should not contain this information at all.

Reply:
I suggest to let the command handler manage if this method needs to be called or not.
An exception could be thrown if one try to remove an image on an entity that does not contain an image.

Reply:
The test is already there https://github.com/EnMarche/en-marche.fr/pull/1927/files#diff-ae8b6be9941e0be84d8476baa425862fR72, just throw a runtime exception.